src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.CoreCLR.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Globalization
```

Comment:
Does this really need to be a separate implementation between CoreCLR and Mono?

Reply:
Not really the only difference is that mono only uses the env var. maybe it doesn’t matter adding the call to AppConfig which will always return false for mono? Didn’t know if it impacted perf and what not.

Reply:
I do not think that the performance is really a concern here. It is a one-time initialization. I would be more worried about functionality differences. We have APIs that let you set the switches if you do it as the first thing before the given sub-system gets initialized.

For example, this prints true on CoreCLR today:
```
using System;
using System.Globalization;

class Program
{
    static void Main(string[] args)
    {
        AppContext.SetSwitch("System.Globalization.Invariant", true);
        Console.WriteLine(CultureInfo.CurrentCulture == CultureInfo.InvariantCulture);
    }
}
```

Reply:
I guess it is fine for people to also be able to set that early on startup right?

So if Mono behaves the same should we unify the code for both to account for AppConfig?

Reply:
Right. I do not see the reason why CoreCLR and Mono should differ.